### PR TITLE
Update README.md of sample-apiserver.

### PR DESCRIPTION
**Pull request by erictune**
Mention CRDs as option.
Link to CRD vs Aggr. docs.
Mention apiserver-builder as alternative.

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/sample-apiserver/README.md [hunk 1]
```diff
@@ -5,7 +5,19 @@ Demonstration of how to use the k8s.io/apiserver library to build a functional A
...
+However, note that there are several other options, all of which are likely better choices than using this repository.
+
+  * If you just want to add a resource to your kubernetes cluster, then you have two choices:
+    * Extension API Server (what this repo implements)
+    * Custom Resource Definition a.k.a CRDs (easier and less coding and rebasing).
+    * Read about the differences between Custom Resource Definitions vs Extension API Servers [here](https://kubernetes.io/docs/concepts/api-extension/custom-resources).
+
+If you do decide to build an Extension API server, consider using [apiserver-builder](https://github.com/kubernetes-incubator/apiserver-builder) instead of this repo, which is a complete framework for generating the both the apiserver and client libraries, and the installation program.
+
+If you do decide to use this repository, then the recommended pattern is to fork this repository, modify it to add your types, and then periodically rebase your changes on top of this repo, to pick up improvements and bug fixes to the apiserver.
```
Comment (nikhita) [inline staging/src/k8s.io/sample-apiserver/README.md]: :+1:

**Comment by nikhita** [comment]
/assign @sttts 

for approval

File: staging/src/k8s.io/sample-apiserver/README.md [hunk 2]
```diff
@@ -5,7 +5,19 @@ Demonstration of how to use the k8s.io/apiserver library to build a functional A
 
 ## Purpose
 
-This is intended as an example of how to build a kube-like API server with a single type.
+You may use this code if you want to build an Extension API Server to use with API Aggregation, or to build a stand-alone Kubernetes-style API server.
+
+
+However, note that there are several other options, all of which are likely better choices than using this repository.
```
Comment (sttts) [inline staging/src/k8s.io/sample-apiserver/README.md]: sounds a bit strange when extensions api server shows up as well below.

File: staging/src/k8s.io/sample-apiserver/README.md [hunk 3]
```diff
@@ -5,7 +5,19 @@ Demonstration of how to use the k8s.io/apiserver library to build a functional A
...
+
+  * If you just want to add a resource to your kubernetes cluster, then you have two choices:
+    * Extension API Server (what this repo implements)
+    * Custom Resource Definition a.k.a CRDs (easier and less coding and rebasing).
+    * Read about the differences between Custom Resource Definitions vs Extension API Servers [here](https://kubernetes.io/docs/concepts/api-extension/custom-resources).
```
Comment (sttts) [inline staging/src/k8s.io/sample-apiserver/README.md]: this is not an option. Should be outdented.

File: staging/src/k8s.io/sample-apiserver/README.md (hunk 2 above)
Comment (sttts) [reply staging/src/k8s.io/sample-apiserver/README.md]: several other options = CRDs, right? So one other option :)

**Comment by erictune** [comment]
@sttts adjusted text per your comments

**Pull request merged**